### ISSUE TITLE
Question: Significance of training with pad_token between every phoneme in the dataset?

### ISSUE THREAD

**Issue by SeymourNickelson** [rhasspy/piper#380]
I noticed that piper places the pad token (index 0) between every phoneme id. In etc/ folder where the test sentences are there is a 0 in between every phoneme id and in the C++ code I see a interspersePad member on the PhonemeIdConfig struct.

I was wondering if someone could explain what is the significance of adding the pad_token between each phoneme? When I try training with a pad_token in between every phoneme id it's taking quite a long time to get started. I get this log repeatedly:

"warning: audio amplitude out of range, auto clipped" and I'm not getting progress on epoch 0. Previously I had a training session going with the same arguments and was a few hundred epochs in (over the course of several days). I'm training from scratch so I was wondering if including the pad_token between each phoneme id in the jsonl file is necessary or could it be omitted? Thanks a lot to anyone who can shed light on this!

**Comment by synesthesiam** [comment]
Interspersing the pad phoneme was something that was found to work well in GlowTTS, which the VITS model in Piper uses. I don't know for sure, but I think this is because it allows the model more flexibility in generating the timings of individual phonemes.
I believe this is no longer used in VITS2, which Piper may upgrade to some day.

**Comment by SeymourNickelson** [comment]
Ah, I see, Thanks a lot for replying.

Maybe it's not that I added the pad_token that's causing my issue. The only other argument I changed from this training session from the previous one is I added --checkpoint-epochs 100 because I read in another issue someone recommended that setting this could speed up training time and I don't need to write every epoch.  Would using --checkpoint-epochs 100 suppress the progress bar from displaying?

**Comment by synesthesiam** [comment]
I don't think so. I've had trouble getting the progress bar to come back with more recent versions of PyTorch Lightning.
If you do set checkpoint epochs to be that high, it may be quite some time before you see a checkpoint.
I'd suggest also checking on progress with tensorboard.

**Comment by SeymourNickelson** [comment]
So I let it run overnight and I got the first checkpoint so training is running but the Terminal isn't showing the progress bar like it was before. I did launch Tensorboard  but it doesn't seem to update the epoch value (for example it shows something like epoch 40 sometimes when I'm actually somewhere in the range between epoch 100 and 198). After refreshing the page a couple times it says epoch 189 (but then when I refresh again sometimes the epoch value goes down?) 

Every time I refresh the page it shows a seemingly random epoch value so it's hard to gauge training progress; not sure if using the --checkpoint-epochs 100 option actually speeds things up in a significant way. I forgot to add the pad token between phonemes in a previous training session so I decided to start over because the results of the already trained models are great and I want to train my model in the same way but was wondering about the significance of using it. I think omitting the pad_token between phonemes in my dataset may boost training time but I'd rather wait for better results.

I'm going to let this training session run but I'm flying semi blind as far as progress goes. Thanks again for answering!
